### DISCUSSION TITLE
Possibility of Jinja2 inside of config context?

### DISCUSSION THREAD

**Discussion by gdprdatasubect** [netbox-community/netbox#14374] [Help Wanted!] [answered]
Hello,

Is it possible to get jinja2 rendered in the config contexts?
For example I wanted a nicely-rendered tenant-name:
``` json
{
    "full-tenant-name": "{{ device.tenant.group.get_family[3] }}-{{ device.tenant }}"
}
```

If that's not possible, is there some other way i can implement the same data without having to replicate it (e.g. custom fields)?

Kind regards

**Comment by candlerb** [comment, marked as answer]
Config contexts are JSON data only. If you want template expansion, that has to be done in the _consumer_ of the config contexts, i.e. whatever application is asking Netbox for the data.  Or you could write a plugin which exposes its own API endpoints.

**Reply by gdprdatasubect** [reply]
Thanks for your answer, do you know if there is kind of a "best-pratice-registry" on how to do certain things on the consumer-side?
Do you think it would be worth it to write an FR for "rendered fields" that contain jinja-rendered data?
